### PR TITLE
Fix #1959 Get message from session

### LINKED ISSUE #1959: XsltSender cannot find [String: TLE/OUTPUTSTUK/RP020.xsl] in scope [nl.nn.adapterframework.senders.XsltSender@22bdb88a]
**Problem**
We are getting the following error.
nl.nn.adapterframework.core.PipeRunException: Pipe [SendToFTConfig_TLE] msgId [5014904] caught exception: IbisLocalSender [SendToFTConfig_TLE] exception calling JavaListener [FTSendersListener_TLE]: Pipe [ROSY_TLE] msgId [5014904] caught exception: IbisLocalSender [ROSY_TLE] exception calling JavaListener [ROSY_TLE]: Pipe [Mapping_TLE_Outputstuk] msgId [5014904] Exception on transforming input: XsltSender [Mapping_TLE_Outputstuk] Exception on transforming input: XsltSender [Mapping_TLE_Outputstuk] Exception on creating transformerHandler chain: XsltSender [Mapping_TLE_Outputstuk] cannot find [String: TLE/OUTPUTSTUK/RP020.xsl] in scope [nl.nn.adapterframework.senders.XsltSender@22bdb88a]

Framework information : 
**FF! 7.7-20210624.162049: ibis4oma ---
Running on LPZWNL000024275 using Apache Tomcat/9.0.39
Java Version: OpenJDK Runtime Environment (1.8.0_252-b09)
Heap size: 263M, total JVM memory: 455M
Free disk space: 99GB, total disk space: 237GB
Up since: 2021-06-25 15:44:41 (2d)**

**Reporter**
Mouli Banerjee

Ali sent me a custom java file <HttpSender.java> **only for local test**. This issue is solved with this custom file.

Waiting for an upgraded framework version to be applied on OTAP.

### PR DESCRIPTION
**Pull request by alisihab**

### CODE REVIEW

File: cmis/src/main/java/nl/nn/adapterframework/extensions/cmis/CmisSender.java [hunk 1]
```diff
@@ -475,7 +475,12 @@ else if (getProperties) {
 	}
 
 	private Message sendMessageForActionCreate(Session cmisSession, Message message, PipeLineSession session) throws SenderException, TimeOutException {
-		String fileName = (String) session.get(getFilenameSessionKey());
+		String fileName = null;
+		try {
+			fileName = session.getMessage(getFilenameSessionKey()).asString();
+		} catch (IOException e) {
+			throw new SenderException("Unable to get filename from pipeline session");
```
Comment (nielsm5) [inline cmis/src/main/java/nl/nn/adapterframework/extensions/cmis/CmisSender.java:482]: This is not entirely correct, `fileName` may be empty. Perhaps you could do a `if(Message.isNotEmpty(...` first before trying to parse a potential `null` as String

File: core/src/main/java/nl/nn/adapterframework/extensions/rekenbox/Adios2XmlPipe.java [hunk 2]
```diff
@@ -335,9 +335,13 @@ protected void initializeConversionTables() throws ConfigurationException	{
...
 	
 	
-	public String findRekenbox(PipeLineSession session) {
+	public String findRekenbox(PipeLineSession session) throws PipeRunException {
 	    if (getRekenboxSessionKey() != null) {
-	        return (String)session.get(getRekenboxSessionKey());
+	        try {
+				return session.getMessage(getRekenboxSessionKey()).asString();
+			} catch (IOException e) {
+				throw new PipeRunException(this, "unable to get rekenbox session key from pipeline session", e);
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/extensions/rekenbox/Adios2XmlPipe.java:343]: I think the error should use the attribute name as a whole `rekenboxSessionKey`

**Review by nielsm5 — APPROVED** [review]

File: core/src/main/java/nl/nn/adapterframework/jdbc/JdbcQuerySenderBase.java [hunk 3]
```diff
@@ -331,10 +331,10 @@ protected PipeRunResult executeStatementSet(QueryExecutionContext queryExecution
 					Object blobSessionVar=null;
 					Object clobSessionVar=null;
 					if (session!=null && StringUtils.isNotEmpty(getBlobSessionKey())) {
-						blobSessionVar=session.get(getBlobSessionKey());
+						blobSessionVar=session.getMessage(getBlobSessionKey()).asString();
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/jdbc/JdbcQuerySenderBase.java:334]: This is not always a String. It is used in JdbcUtil.streamBlob()

File: core/src/main/java/nl/nn/adapterframework/jdbc/JdbcQuerySenderBase.java [hunk 4]
```diff
@@ -331,10 +331,10 @@ protected PipeRunResult executeStatementSet(QueryExecutionContext queryExecution
...
 					}
 					if (session!=null && StringUtils.isNotEmpty(getClobSessionKey())) {
-						clobSessionVar=session.get(getClobSessionKey());
+						clobSessionVar=session.getMessage(getClobSessionKey()).asString();
```
Comment (gvanbrakel) [inline core/src/main/java/nl/nn/adapterframework/jdbc/JdbcQuerySenderBase.java:337]: This is not always a String. It is used in JdbcUtil.streamClob()

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

**Review by gvanbrakel — APPROVED** [review]

**Pull request merged**